R/subplots.R

```diff
@@ -290,8 +291,8 @@ subplot <- function(..., nrows = 1, widths = NULL, heights = NULL, margin = 0.02
     }
     layouts <- layouts[which_layout]
   }
-  p$attrs <- Reduce(c, lapply(plots, "[[", "attrs"))
-  p$layout <- modify_list(p$layout, Reduce(modify_list, layouts))
+  p$attrs <- unlist(lapply(plots, "[[", "attrs"), recursive = FALSE)
+  p$layout <- Reduce(modify_list, layouts, p$layout)
```

Comment:
If there are a lot of layouts then `do.call(modify_list, c(list(p$layout), layouts))` might be better (or something like that with judicious `quote()` calls added--@wch can give you details if you choose to go the `do.call` route). If there aren't a lot of layouts then this is fine with me.